src/solver/CMakeLists.txt

```diff
@@ -1,22 +1,15 @@
-# =============================================================================
-# Copy necessary files to build directory
-# =============================================================================
-
-cpp_cc_build_time_copy(INPUT "${CMAKE_CURRENT_SOURCE_DIR}/newton/newton.hpp" OUTPUT
-                       "${CMAKE_BINARY_DIR}/include/newton/newton.hpp")
-
-cpp_cc_build_time_copy(INPUT "${CMAKE_CURRENT_SOURCE_DIR}/crout/crout.hpp" OUTPUT
-                       "${CMAKE_BINARY_DIR}/include/crout/crout.hpp")
-
-add_custom_target(
-  nmodl_copy_solver_files ALL DEPENDS "${CMAKE_BINARY_DIR}/include/newton/newton.hpp"
-                                      "${CMAKE_BINARY_DIR}/include/crout/crout.hpp")
-
-# Eigen
-file(COPY ${NMODL_PROJECT_SOURCE_DIR}/ext/eigen/Eigen DESTINATION ${CMAKE_BINARY_DIR}/include/)
-
-# =============================================================================
-# Install solver headers and eigen from include
-# =============================================================================
-
-install(DIRECTORY ${CMAKE_BINARY_DIR}/include/ DESTINATION ${NMODL_INSTALL_DIR_SUFFIX}include)
+# Read headers, remove everything up to and including "#pragma once", then remove header includes.
+file(READ ${CMAKE_CURRENT_SOURCE_DIR}/crout/crout.hpp NMODL_CROUT_HPP_RAW)
+string(REGEX REPLACE ".*#pragma once[ \t\r\n]*" "" NMODL_CROUT_HPP "${NMODL_CROUT_HPP_RAW}")
+file(READ ${CMAKE_CURRENT_SOURCE_DIR}/newton/newton.hpp NMODL_NEWTON_HPP_RAW)
+string(REGEX REPLACE ".*#pragma once[ \t\r\n]*" "" NMODL_NEWTON_HPP_TMP "${NMODL_NEWTON_HPP_RAW}")
+string(REGEX REPLACE "#include <crout/crout.hpp>[ \t\r\n]*" "" NMODL_NEWTON_HPP
+                     "${NMODL_NEWTON_HPP_TMP}")
```

Comment:
Like this PR! The only thing I am a bit "uncomfortable" about is this string processing in CMake and I wonder if there could be a better / more robust way?

Let's see also what @alkino says!

Reply:
in CMake there is none

Reply:
I decided to leave the headers as standard C++ - if I wanted to forego the CMake text manipulation, I'd have to at least remove the `#pragma once`  - which seems a step backwards.

Reply:
Yes, in the GitLab CI we test a full installation of Neurodamus with CoreNeuron and NMODL.

This is the same guard we have against typos in the code generation, too, no?

Reply:
Ok. I just thinking of gitlab going away and if this would be tested locally or GitHub.

 I took a look at the current CI configs and it's true that we have been only testing code generation and not compiling/linking.

But, very soon `test/usecases` will have solver related mod files and then this will be tested as part of github CI. I would say this is good to go!